sdk/java/src/main/java/com/chain/api/PagedItems.java

```diff
@@ -105,4 +112,15 @@ public boolean hasNext() {
       }
     }
   }
+
+  /**
+   * Removes the last element returned by {@link PagedItems#next()} from the underlying collection.
+   */
+  public void remove() throws IllegalStateException {
+    if (!canRemove) {
```

Comment:
TL;DR we should consider simplifying this function to `throw UnsupportedOperationException`.

According to the [Java 7 docs](https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html), it is perfectly fine to implement this method by throwing `UnsupportedOperationException`. For our case, I believe this is the more appropriate behavior. `remove` makes sense if the iterator object refers back to a collection that the user interacts with directly. In other words, it seems to be appropriate for this type of situation:

```
Collection c;
Iterator i = c.iterator();
while (i.hasNext()) {
  Object o = i.next();
  if (shouldRemove(0)) {
    i.remove();
  }
}

// Now collection c is updated.
```

However, I believe `PagedItems` works a bit differently. While there is an underlying `List<T>`, we don't really encourage using that list directly. Calling `remove` on a `PagedItems` object is a weird use case.

Moreover, it may be better for us to make `list` a protected member, so that users are not tempted to use it directly.